ale/transformation.py

```diff
@@ -128,34 +128,33 @@ def from_spice(cls, sensor_frame, target_frame, center_ephemeris_time,
 
         constant_frames.extend(target_constant_frames)
 
-        sensor_times = []
-        if exact_ck_times and len(ephemeris_times) > 1 and not nadir and not frame_chain.use_web:
+        # Do this call so we know if we can get exact ck times for our data
+        if exact_ck_times and len(ephemeris_times) > 1 and not nadir:
             try:
-                sensor_times = spiceql_call("extractExactCkTimes", {"observStart": ephemeris_times[0] + inst_time_bias, 
-                                                                    "observEnd": ephemeris_times[-1] + inst_time_bias, 
-                                                                    "targetFrame": sensor_frame,
-                                                                    "mission": mission,
-                                                                    "searchKernels": frame_chain.search_kernels},
-                                                                    use_web=frame_chain.use_web)
+                times = spiceql_call("extractExactCkTimes", {"observStart": ephemeris_times[0] + inst_time_bias, 
+                                                             "observEnd": ephemeris_times[-1] + inst_time_bias, 
+                                                             "targetFrame": sensor_frame,
+                                                             "mission": mission,
+                                                             "searchKernels": frame_chain.search_kernels},
+                                                             use_web=frame_chain.use_web)
+
+                if len(times) == 0:
+                    exact_ck_times = False
+
             except Exception as e:
+                exact_ck_times = False
```

Comment:
Similar to my other comment, Im not sure we want to spend time with the extra call if web==True. 

There might be a concern that if exact_times==False and web==True, and the sensor is a linescanner, things will take forever.

Reply:
Sure, the extra call here is to handle if there are multiple cks that are loaded for the image. If the function fails, then we can't get exact ck times so we set `exact_ck_times` to false.

This then goes to the rotation generation that will run `getTargetOrientations` rather than `getExactTargetOrientations`. So you're right it may take a while for large line scanners. I think we need to address other ways of reducing those times rather than adding `if else` blocks that all over ALE to handle specific cases.